### DISCUSSION TITLE
posts/enable-google-pv/

### DISCUSSION THREAD

**Discussion by giscus** [cotes2020/chirpy-demo#10] [Comments]
# posts/enable-google-pv/

This post is to enable Page Views on the Chirpy theme based blog that you just built. This requires technical knowledge and it’s recommended to keep the google_analytics.pv.* empty unless you have a good reason. If your website has low traffic, the page views count would discourage you to write more blogs. With that said, let’s start with the setup.

https://chirpy.cotes.page/posts/enable-google-pv/

**Comment by yywe** [comment]
It seems that Google Analytics superProxy does not support GA4. https://github.com/googleanalytics/google-analytics-super-proxy/issues/50, it requires view ID. 

And currently  GA4 adopted a new data model without 'view'.  in  https://ga-dev-tools.web.app/query-explorer/, if we use GA4, there is not URI generated. 

do we have any other solution to add page view?

Thanks!

**Reply by JackieZhai** [reply]
You may follow the instructions here:
https://vansoest.dev/posts/creating-a-dev-blog/#cdn-and-analytics--google-cloud

But unfortunately, Google said Universal Analytics for GA4 would be only supported until 2023-07-01...

**Reply by cotes2020** [reply]
@yywe So far, no open source way has been found to provide the PV report of _GA 4_. If there are new clues, please let me know.

**Comment by tired-o** [comment]
It took few hours to enable page views but it's worthy. Thank you!

**Comment by tazihad** [comment]
@cotes2020 you found the new way to add page views?

**Comment by abmounir** [comment]
thanks

**Comment by aouledissa** [comment]
Please update the docs and source code to support GA 4. The current setup doesn't work

**Reply by cotes2020** [reply]
May I ask what makes you think that _Chirpy_ doesn't support GA4? The prompt at the top of this post says: _surperProxy_ can't fetch pageviews from GA4, it doesn't mean that Chirpy can't use GA4, you just need to put  GA4 ID in the configuration like UA.

**Reply by aouledissa** [reply]
I tried following the steps you have provided but the GA4 was not registering anything for over 48h. So i went ahead and ijected the `gtag` manually as a header for each page. You can follow up my implementation here https://aouledissa.com/posts/Jekyll-Google-Analytics-4-Integration-With-Chirpy-Theme/

**Reply by cotes2020** [reply]
I just read the link you provided and the ["Setup Google Analytics"](https://aouledissa.com/posts/Jekyll-Google-Analytics-4-Integration-With-Chirpy-Theme/#setup-google-analytics-%EF%B8%8F) section on how to create GA 4 ID is fine. But the section ["Configure gtag script globally"](https://aouledissa.com/posts/Jekyll-Google-Analytics-4-Integration-With-Chirpy-Theme/#configure-gtag-script-globally-) is totally unnecessary and even misleading.

The `gtag` code snippet has existed in [3c2675c](https://github.com/cotes2020/jekyll-theme-chirpy/commit/3c2675c4a154871b572d57cfd4b90f26652e733b) since 12/10/2020.

**Reply by aouledissa** [reply]
the first [section](https://aouledissa.com/posts/Jekyll-Google-Analytics-4-Integration-With-Chirpy-Theme/#setup-google-analytics-%EF%B8%8F) is no different than what you have in your own blog posting. However when i tried setting up the GA4 id in the `_config.yml` file I saw no traffic coming as  if there was no tracking at all. When I setup the `gtag` as I have explained, I immediately started seeing PVs with detailed data (country, users, page view etc..) in the GA4 dashboard. I am not sure I understand why you say it's unnecessary. Could you please detail more?

**Reply by cotes2020** [reply]
1. You removed the dynamic code for the `gtag` and replaced it with hardcoding, but both are the same.

2. The deleted code contains logic that optimizes for web performance and therefore reduces the site page load performance, so I would say it's misleading to others.

**Reply by aouledissa** [reply]
Thanks for the explanation! I have one question still:
- Why the original code didn't work with the just configuring the GA4 ID?

**Reply by cotes2020** [reply]
There are various situations where _GA_ does not collect data, sometimes it is blocked by plugins, sometimes there is a data delay that takes hours to days, you should refer to Google Docs for the relevant explanation.

When using _Chirpy_, all you need to ensure is that the built HTML page contains a `gtag` code snippet with your GA 4 id.

**Reply by aouledissa** [reply]
I will revert the code and debug further on my end! If you're interested I will open an issue about this. Thanks for the explanation once again!

**Reply by cotes2020** [reply]
To be honest, I'm not too interested in it because the code snippet is simple enough and has been collecting GA4 data for over 1 year on this demo site.  :)

**Reply by aouledissa** [reply]
No worries! That's fine for me. I will go this route my self simply because it's not collecting any data even after days.

**Reply by cotes2020** [reply]
As long as you meet the conditions of [this line](https://github.com/cotes2020/jekyll-theme-chirpy/blob/b63b65d74321c43001c6cdb5de3ed58b8998fdf8/_includes/head.html#L73), I believe GA can work properly.

**Comment by aouledissa** [comment]
If you're looking into a way to integrate GA 4 pvs with Chirpy, I have posted a new blog post detailing the integration process 😁. Check it out here https://aouledissa.com/posts/Jekyll-Google-Analytics-4-Integration-With-Chirpy-Theme/

**Reply by cordovez** [reply]
Thank you Mohamed. Very clear explanation.

**Reply by aouledissa** [reply]
You're welcome @cordovez

**Reply by sukhjinder-kumar** [reply]
I had a doubt in your blog post, there was no comment there so asking here. How to show page view on top of blog page? Preferably without credit card information as given here.

Thanks and great blog post!
